src/shared/tpm2-util.c

```diff
@@ -5616,12 +5644,11 @@ int tpm2_unseal(Tpm2Context *c,
 
         usec_t start = now(CLOCK_MONOTONIC);
 
-        TPM2B_PUBLIC public;
-        TPM2B_PRIVATE private;
-        TPM2B_ENCRYPTED_SECRET seed = {};
-        r = tpm2_unmarshal_blob(blob->iov_base, blob->iov_len, &public, &private, &seed);
-        if (r < 0)
-                return log_debug_errno(r, "Could not extract parts from blob: %m");
+        size_t n_shards = pcrlock_policy && iovec_is_set(pubkey) ? 2 : 1;
+        if (n_blobs != n_shards)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Number of provided key blobs (%zu) does not match policy requirements (%zu).", n_blobs, n_shards);
+        if (n_known_policy_hash > 0 && n_known_policy_hash != n_shards)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Number of provided known policy hashes (%zu) does not match policy requirements (%zu or 0).", n_known_policy_hash, n_shards);
```

Comment:
This function does not provide log message higher than LOG_DEBUG. Please downgrade them to `log_debug_errno()`.